src/backend/src/services/work-packages.services.ts

```diff
@@ -701,4 +702,100 @@ export default class WorkPackagesService {
 
     return workPackageTemplateTransformer(workPackage);
   }
+
+  static async editWorkPackageTemplate(
+    user: User,
+    workPackageTemplateId: string,
+    templateName: string,
+    templateNotes: string,
+    duration: number | undefined,
+    stage: WorkPackageStage | undefined,
+    blockedBy: BlockedByInfo[],
+    expectedActivities: string[],
+    deliverables: string[],
+    workPackageName: string | undefined
+  ): Promise<Work_Package_Template> {
+    // retrieving original work package template
+    const originalWorkPackageTemplate = await prisma.work_Package_Template.findUnique({
+      where: { workPackageTemplateId },
+      include: { blockedBy: true }
+    });
+
+    // error handling
+    if (!originalWorkPackageTemplate) throw new NotFoundException('Work Package Template', workPackageTemplateId);
+    if (originalWorkPackageTemplate.dateDeleted) throw new DeletedException('Work Package Template', workPackageTemplateId);
+    if (!isAdmin(user.role)) throw new AccessDeniedGuestException('edit work package templates');
+
+    // retrieving list of blocked By ids for work package template
+    const blockedByIds = originalWorkPackageTemplate.blockedBy.map((item) => item.blockedByInfoId);
```

Comment:
I think some things are getting confused in this section because this list is being made from the original, but some places are acting as if this is the new list